src/network/networkd-ndisc.c

```diff
@@ -1029,6 +1029,69 @@ static int ndisc_drop_outdated(Link *link, usec_t timestamp_usec) {
         return r;
 }
 
+static int ndisc_process_zero_lifetime(Link *link, sd_ndisc_router *rt) {
```

Comment:
This functions seems to be almost a copy of `ndisc_drop_outdated()`. Instead of duplicating this logic here, just call `ndisc_drop_outdated(timestamp_usec=INFINITY)`. (Maybe some additional conditionalization of `ndisc_drop_outdated()` will be needed… But it's certainly better than having two functions doing almost the same thing.

Reply:
Thanks for taking a look. We need the ` sd_ndisc_router` object. There could be a multiple routers in the network with diff precedence.  `ndisc_drop_outdated` does not have this object.  We are comparing with the `provider`  i.e. from which source this resource got configured. and dropping.